spec/support/retry.cr

```diff
@@ -0,0 +1,17 @@
+def retry(n = 5)
+  exception = nil
+  n.times do |i|
+    yield
+  rescue ex
+    exception = ex
+    if i == 0
+      Fiber.yield
+    else
+      sleep 0.01 * (2**i)
+    end
+  else
+    return
+  end
+
+  raise exception.not_nil!
```

Comment:
Why not just:

```suggestion
  n.times do |i|
    return yield
  rescue ex
    if i == 0
      Fiber.yield
    else
      sleep 0.01 * (2**i)
    end
    raise ex
  end
end
```

Reply:
Because is not the same. The `return yield` might work though.